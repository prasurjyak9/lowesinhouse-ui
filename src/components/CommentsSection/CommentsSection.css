.previous-commentsbox{
    width: 830px;
    margin-top: 2%;
    color:black;
    
    
}
.mainGrid
{
    background-color: whitesmoke;
    border-radius:5px;
}
.mainCommentSection
{
    border-radius: 50%;
    margin-top: 2%;
}
.addCommentButton
{
    margin-bottom: 10px;
}