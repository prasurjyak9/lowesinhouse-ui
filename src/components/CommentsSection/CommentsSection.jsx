import { useState, useEffect } from "react";

function CommentsSection() {
    const [comments, setComments] = useState([])
    const [newComment, setNewComment] = useState('')
    const [inputValue, setInputValue] = useState('')

    useEffect(() => {
        setComments(["first comment", "second comment", "third comment"])
    }, [])

    function handleCommentInput(e) {
        setNewComment(e.target.value)
        setInputValue(e.target.value)
    }

    function handleAddComment(e) {
        setComments([...comments, newComment])
        setInputValue('')
    }

    return (
        <div>
            <div className="previous-comments">
                {comments.map((cmt, idx) => <p>{cmt}</p>)}
            </div>
            <div className="add-comment">
                <input type="text" onChange={handleCommentInput} value={inputValue}></input>
                <button onClick={handleAddComment}>Add Comment</button>
            </div>
        </div>
    )
}

export default CommentsSection;
