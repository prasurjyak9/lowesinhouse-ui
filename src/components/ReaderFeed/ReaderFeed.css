.readersfeeddiv{
    
   background-color: #01216a;
    
  }