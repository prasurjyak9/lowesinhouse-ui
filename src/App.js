import React from 'react';
import { Routes, Route } from 'react-router-dom'
import EditableCanvas from './components/Editor/Editor';
import NewsLetter from './components/Newsletter/Newsletter';
import ReaderFeed from './ReaderFeed/ReaderFeed';


function App() {
  return (
    <div>
      <Routes>
        <Route path="/" element={<EditableCanvas />}></Route>
        <Route path="/read/" element={<NewsLetter />}></Route>
        <Route path="/feed/" element={<ReaderFeed />}></Route>
      </Routes>
    </div>
  )
}

export default App;