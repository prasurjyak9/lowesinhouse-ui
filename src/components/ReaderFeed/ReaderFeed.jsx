import React, { useState, useEffect } from "react";
import NewsLetter from "../Newsletter/Newsletter";
import "./ReaderFeed.css"
import Navbar from "../Navbar/Navbar";
import axios from "axios";
import { Link } from 'react-router-dom';

function ReaderFeed() {
    const [newsFeed, setNewsFeed] = useState([])

    useEffect(() => {
        let url = "http://localhost:8080/"

        axios
            .get(url)
            .then(response => {
                console.log("res = ", response.data)
                setNewsFeed(response.data)
            })
    }, [])


    return (
        <div className="readersfeeddiv">
            <Link to={"login"} state={{ role: "designer" }}>
                Designer
            </Link>
            <Link to={"login"} state={{ role: "approver" }}>
                Approver
            </Link>
            {newsFeed.map((nl, idx) => <NewsLetter key={idx} content={nl.content} id={"canvas" + idx} comments={nl.remarks} key={"canvas" + idx} />)}
        </div>
    )
}

export default ReaderFeed;