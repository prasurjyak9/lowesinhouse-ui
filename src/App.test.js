import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';
import EditableCanvas from './components/Editor/Editor';
import {create} from 'react-test-renderer'
import NewsLetter from './components/Newsletter/Newsletter';
import CommentsSection from './components/CommentsSection/CommentsSection';
import ReaderFeed from './components/ReaderFeed/ReaderFeed';


// test('test scenario 1', () => {
//   render(<EditableCanvas />)
//   const element = screen.getByText("Export");
//   expect(element).toBeInTheDocument();
// });

// test('test scenario 2', () => {
//   render(<EditableCanvas />)
//   const element = screen.getByText("Add Text");
//   expect(element).toBeInTheDocument();
// });

// test('test scenario 3', () => {
//   render(<EditableCanvas />)
//   const element = screen.getByText("Add Image With Url");
//   expect(element).toBeInTheDocument();
// });

// test('test scenario 4', () => {
//   render(<EditableCanvas />)
//   const btn = screen.getByText('Add Text')
//   console.log("btn =", btn)
//   fireEvent.click(btn)
  
// });

// test('test scenario 5', () => {
//   render(<EditableCanvas />)
//   const btn = screen.getByText('Add Text')
//   console.log("btn =", btn)
//   fireEvent.click(btn)
  
// });

test('test scenario 6', () => {
  let tree = create(<EditableCanvas />)
  expect(tree.toJSON()).toMatchSnapshot();
})

test('test scenario 7', () => {
  let tree = create(<NewsLetter />)
  expect(tree.toJSON()).toMatchSnapshot();
})

test('test scenario 8', () => {
  let tree = create(<CommentsSection />)
  expect(tree.toJSON()).toMatchSnapshot();
})

test('test scenario 9', () => {
  let tree = create(<ReaderFeed />)
  expect(tree.toJSON()).toMatchSnapshot();
})

// test('test scenario 10', () => {
//   render(<CommentsSection />)
//   const element = screen.getByText("Add Comment");
//   expect(element).toBeInTheDocument();
// });

// test('test scenario 11', () => {
//   render(<CommentsSection />)
//   const inputFields = screen.getAllByRole("textbox")
//   fireEvent.change(inputFields[0], { target: { value: 4 } })

//   const element = screen.getByRole('button')
//   fireEvent.click(element)

//   const comments = screen.getAllByRole("paragraph")
//   expect(comments[3]).toHaveValue("4")
// })