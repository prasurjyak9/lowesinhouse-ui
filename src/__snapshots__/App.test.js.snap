// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test scenario 6 1`] = `
Array [
  <div>
    <canvas
      id="canvas"
    />
    <button
      onClick={[Function]}
    >
      Add Text
    </button>
    <button
      onClick={[Function]}
    >
      Import
    </button>
    <button
      onClick={[Function]}
    >
      Export
    </button>
  </div>,
  <form>
    <label>
      Enter Image Url here : 
    </label>
    <input
      id="input"
    />
    <button
      onClick={[Function]}
      type="button"
    >
      Add Image With Url
    </button>
  </form>,
  <a
    download=""
    href=""
    onClick={[Function]}
  >
    Print As Image
  </a>,
  <button
    onClick={[Function]}
  >
    Show as HTML
  </button>,
]
`;

exports[`test scenario 7 1`] = `
<div>
  <canvas />
  <div>
    <div
      className="previous-comments"
    />
    <div
      className="add-comment"
    >
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        onClick={[Function]}
      >
        Add Comment
      </button>
    </div>
  </div>
</div>
`;

exports[`test scenario 8 1`] = `
<div>
  <div
    className="previous-comments"
  />
  <div
    className="add-comment"
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Add Comment
    </button>
  </div>
</div>
`;

exports[`test scenario 9 1`] = `
<div>
  <div>
    <canvas
      id="canvas0"
    />
    <div>
      <div
        className="previous-comments"
      />
      <div
        className="add-comment"
      >
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
        <button
          onClick={[Function]}
        >
          Add Comment
        </button>
      </div>
    </div>
  </div>
  <div>
    <canvas
      id="canvas1"
    />
    <div>
      <div
        className="previous-comments"
      />
      <div
        className="add-comment"
      >
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
        <button
          onClick={[Function]}
        >
          Add Comment
        </button>
      </div>
    </div>
  </div>
</div>
`;
